### ISSUE TITLE
Require await when calling "gen" prefixed functions

### ISSUE THREAD

**Issue by mhagmajer** [babel/babel-eslint#320]
At Facebook I came across a convention to prefix all functions that return `Promise` with `gen`. So that it's immediately clear to a developer that they need to await it:

```js
await this.genDoSomething();
```

The reason why this is important is that if you forget `await` then (while having type-valid code):
a) execution of async block is not suspended;
b) the method may fail silently while still being type-correct.

It would be great to have an eslint rule to enforce this. I'd be willing to help with creating one. Can someone knowledgable say if this is feasible and how can I go about this?

**Comment by hzoo** [comment]
babel-eslint doesn't come with any rules so you wouldn't be making a PR here. If we accept it, it would go in https://github.com/babel/eslint-plugin-babel, although it seems to be a bit specific and maybe better as a custom rule?

You can definetely check that an Identifier in the CallExpression starts with `gen` but there's no way to know it returns a Promise or if other methods to start with gen as well.

**Comment by mhagmajer** [comment]
Thanks @hzoo I'd like this rule to only check that all `gen`-prefixed functions are awaited. That I can do, right?

**Comment by hzoo** [comment]
Sorry, yeah for sure (just not this in this particular repo)

**Comment by mhagmajer** [comment]
Thanks @hzoo !
